src/sagemaker/session.py

```diff
@@ -763,7 +763,10 @@ def get_caller_identity_arn(self):
 
         # Call IAM to get the role's path
         role_name = role[role.rfind('/') + 1:]
-        role = self.boto_session.client('iam').get_role(RoleName=role_name)['Role']['Arn']
+        try:
+            role = self.boto_session.client('iam').get_role(RoleName=role_name)['Role']['Arn']
+        except ClientError:
+            LOGGER.warning("Couldn't call 'get_role' to get Role ARN from role name {}.".format(role_name))
```

Comment:
I'd include a note here that this will exclude a role's path if it has one

Reply:
@andremoeller @laurenyu 
*AmazonSageMakerFullAccess* that is a managed policy for SageMaker \(see [doc](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-amazonsagemakerfullaccess-policy)\) does not allow `iam:GetRole`.  The line in `try`always fail and goes to `except` if users attach only *AmazonSageMakerFullAccess* to the role; `get_role()` does not work. Could you consider another implementation for this?

Reply:
@harusametime 

You're right -- in that case. The purpose of this is to attempt to get the role using `iam:GetRole`, but never to fail in this case -- not all users will only be using `AmazonSageMakerFullAccess` policy.